test/core.jl

```diff
@@ -4558,3 +4558,15 @@ function f18173()
     successflag = false
 end
 @test f18173() == false
+
+let _true = Ref(true), f, g, h
+    @noinline f() = ccall((:error, :error), Void, ()) # some expression that throws an error in codegen
```

Comment:
isn't this fragile if someone happens to have an `error.so` somewhere on their library path?